### PR TITLE
bugfix: getJobConfiguration return null instead of throw NullPointerException

### PR DESCRIPTION
**Pull request by stanstanwang**
Bugfix: getJobConfiguration return null instead of throw NullPointerException

Changes proposed in this pull request:
- check the zk result before `YamlEngine.unmarshal`

Before that, getting the JobConfiguration was more troublesome
``` java
JobConfigurationPOJO jobConfiguration = null;
try {
    jobConfiguration = jobConfigurationAPI.getJobConfiguration(jobKey);
} catch (NullPointerException e) {
    // ignore
}
// do something
```

### CODE REVIEW

**Review by TeslaCN — CHANGES_REQUESTED** [review]
Hi @stanstanwang 
I think returning null is not elegant. How about throw something like `JobConfigurationException`?

File: elasticjob-lite/elasticjob-lite-lifecycle/src/main/java/org/apache/shardingsphere/elasticjob/lite/lifecycle/internal/settings/JobConfigurationAPIImpl.java [hunk 1]
```diff
@@ -31,22 +31,26 @@
  */
 @RequiredArgsConstructor
 public final class JobConfigurationAPIImpl implements JobConfigurationAPI {
-    
+
```
Comment (TeslaCN) [inline elasticjob-lite/elasticjob-lite-lifecycle/src/main/java/org/apache/shardingsphere/elasticjob/lite/lifecycle/internal/settings/JobConfigurationAPIImpl.java]: BTW, keep the indents please.

**Comment by TeslaCN** [comment]
Code of conduct: https://shardingsphere.apache.org/community/en/contribute/code-conduct/

File: elasticjob-lite/elasticjob-lite-lifecycle/src/main/java/org/apache/shardingsphere/elasticjob/lite/lifecycle/internal/settings/JobConfigurationAPIImpl.java (hunk 1 above)
Comment (stanstanwang) [reply elasticjob-lite/elasticjob-lite-lifecycle/src/main/java/org/apache/shardingsphere/elasticjob/lite/lifecycle/internal/settings/JobConfigurationAPIImpl.java]: fixed，sorry but there is a litte strange why  4 space indent in every empty line

**Comment by stanstanwang** [comment]
> Hi @stanstanwang
> I think returning null is not elegant. How about throw something like `JobConfigurationException`?

in my usage scenario , there is no check config exists method, so every time before i  get the config, i must catch the NullPointerException , this  is troublesome,  may be mark the method return @Nullable or using Optional is also elegant

**Comment by Technoboy-** [comment]
> > Hi @stanstanwang
> > I think returning null is not elegant. How about throw something like `JobConfigurationException`?
> 
> in my usage scenario , there is no check config exists method, so every time before i get the config, i must catch the NullPointerException , this is troublesome, may be mark the method return @nullable or using Optional is also elegant

Currently, fit it that not throw NPE is enough. Other further discuss can open a new thread to discuss. @TeslaCN

**Comment by TeslaCN** [comment]
Returning `null` is violate our code of conduct. I think `Optional` may be better.

**Comment by Technoboy-** [comment]
> Returning `null` is violate our code of conduct. I think `Optional` may be better.

Where do this conflict?  If conflict, we should also refactor ZookeeperRegistryCenter. Not this API.

**Comment by TeslaCN** [comment]
> > Returning `null` is violate our code of conduct. I think `Optional` may be better.
> 
> Where do this conflict? If conflict, we should also refactor ZookeeperRegistryCenter. Not this API.

This API is used by user directly.  Maybe the `ZookeeperRegistryCenter` does need to be refactored but not in this issue.

**Comment by Technoboy-** [comment]
> > > Returning `null` is violate our code of conduct. I think `Optional` may be better.
> > 
> > 
> > Where do this conflict? If conflict, we should also refactor ZookeeperRegistryCenter. Not this API.
> 
> This API is used by user directly. Maybe the `ZookeeperRegistryCenter` does need to be refactored but not in this issue.

If user upgrade to current version, compile error in they project.

File: elasticjob-lite/elasticjob-lite-lifecycle/src/main/java/org/apache/shardingsphere/elasticjob/lite/lifecycle/internal/settings/JobConfigurationAPIImpl.java [hunk 2]
```diff
@@ -36,7 +36,11 @@
     
     @Override
     public JobConfigurationPOJO getJobConfiguration(final String jobName) {
-        return YamlEngine.unmarshal(regCenter.get(new JobNodePath(jobName).getConfigNodePath()), JobConfigurationPOJO.class);
+        String yamlContent = regCenter.get(new JobNodePath(jobName).getConfigNodePath());
+        if (yamlContent == null) {
```
Comment (TeslaCN) [inline elasticjob-lite/elasticjob-lite-lifecycle/src/main/java/org/apache/shardingsphere/elasticjob/lite/lifecycle/internal/settings/JobConfigurationAPIImpl.java]: Constant is on the left.

**Comment by TeslaCN** [comment]
> > > > Returning `null` is violate our code of conduct. I think `Optional` may be better.
> > > 
> > > 
> > > Where do this conflict? If conflict, we should also refactor ZookeeperRegistryCenter. Not this API.
> > 
> > 
> > This API is used by user directly. Maybe the `ZookeeperRegistryCenter` does need to be refactored but not in this issue.
> 
> If user upgrade to current version, compile error in they project.

OK, we can keep the current API. But we should polish it in future.

File: elasticjob-lite/elasticjob-lite-lifecycle/src/main/java/org/apache/shardingsphere/elasticjob/lite/lifecycle/internal/settings/JobConfigurationAPIImpl.java (hunk 2 above)
Comment (stanstanwang) [reply elasticjob-lite/elasticjob-lite-lifecycle/src/main/java/org/apache/shardingsphere/elasticjob/lite/lifecycle/internal/settings/JobConfigurationAPIImpl.java]: fixed, sorry for that..

File: elasticjob-lite/elasticjob-lite-lifecycle/src/main/java/org/apache/shardingsphere/elasticjob/lite/lifecycle/internal/settings/JobConfigurationAPIImpl.java [hunk 3]
```diff
@@ -36,7 +36,11 @@
     
     @Override
     public JobConfigurationPOJO getJobConfiguration(final String jobName) {
-        return YamlEngine.unmarshal(regCenter.get(new JobNodePath(jobName).getConfigNodePath()), JobConfigurationPOJO.class);
+        String yamlContent = regCenter.get(new JobNodePath(jobName).getConfigNodePath());
```
Comment (Technoboy-) [inline elasticjob-lite/elasticjob-lite-lifecycle/src/main/java/org/apache/shardingsphere/elasticjob/lite/lifecycle/internal/settings/JobConfigurationAPIImpl.java:39]: How about like this:
Optional.ofNullable(yamlContent).map(v -> YamlEngine.unmarshal(v, JobConfigurationPOJO.class)).orElse(null) ?
@TeslaCN @stanstanwang

**Review by Technoboy- — APPROVED** [review]
LGTM

**Review by TeslaCN — APPROVED** [review]

**Pull request merged**